Failed to iterate over archive on Windows

```log
2025-03-18T18:38:58.8755489+01:00 [ERROR] Failed to start language server "code-stats-ls": failed to download file: failed to iterate over archive
2025-03-18T18:38:58.8756239+01:00 [ERROR] server stderr: ""
```

```
Zed: v0.177.9 (Zed) 
OS: Windows 10.0.22631
Memory: 63.1 GiB
Architecture: x86_64
GPU: NVIDIA GeForce RTX 4090 || NVIDIA || 572.60
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -12,7 +12,13 @@ impl CodeStatsExtension {
         language_server_id: &LanguageServerId,
         worktree: &Worktree,
     ) -> Result<String> {
-        if let Some(path) = worktree.which("code-stats-ls") {
+        let (platform, arch) = zed::current_platform();
+        let binary_name = match platform {
+            zed::Os::Mac | zed::Os::Linux => "code-stats-ls",
+            zed::Os::Windows => "code-stats-ls.exe",
+        };
+
+        if let Some(path) = worktree.which(&binary_name) {
             return Ok(path);
         }
 
@@ -34,7 +40,6 @@ impl CodeStatsExtension {
             },
         )?;
 
-        let (platform, arch) = zed::current_platform();
         let target_triple = format!(
             "code-stats-ls-{arch}-{os}",
             arch = match arch {
@@ -64,7 +69,13 @@ impl CodeStatsExtension {
             .ok_or_else(|| format!("no asset found matching {:?}", asset_name))?;
 
         let version_dir = format!("code-stats-ls-{}", release.version);
-        let binary_path = format!("{version_dir}/{target_triple}/code-stats-ls");
+
+        // Windows have a different archive structure, as documented at:
+        // https://axodotdev.github.io/cargo-dist/book/artifacts/archives.html#archive-contents
+        let binary_path = match platform {
+            zed::Os::Windows => format!("{version_dir}/{}", binary_name),
+            _ => format!("{version_dir}/{target_triple}/{}", binary_name),
+        };
 
         if !fs::metadata(&binary_path).map_or(false, |stat| stat.is_file()) {
             zed::set_language_server_installation_status(
@@ -72,10 +83,15 @@ impl CodeStatsExtension {
                 &zed::LanguageServerInstallationStatus::Downloading,
             );
 
+            let download_type = match platform {
+                zed::Os::Windows => zed::DownloadedFileType::Zip,
+                _ => zed::DownloadedFileType::GzipTar,
+            };
+
             zed::download_file(
                 &asset.download_url,
                 &version_dir,
-                zed::DownloadedFileType::GzipTar,
+                download_type,
             )
             .map_err(|err| format!("failed to download file: {err}"))?;
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -64,7 +64,6 @@ impl CodeStatsExtension {
             .ok_or_else(|| format!("no asset found matching {:?}", asset_name))?;
 
         let version_dir = format!("code-stats-ls-{}", release.version);
-
         let (binary_path, file_type) = match platform {
             zed::Os::Mac | zed::Os::Linux => (
                 format!("{version_dir}/{target_triple}/code-stats-ls"),

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -12,13 +12,7 @@ impl CodeStatsExtension {
         language_server_id: &LanguageServerId,
         worktree: &Worktree,
     ) -> Result<String> {
-        let (platform, arch) = zed::current_platform();
-        let binary_name = match platform {
-            zed::Os::Mac | zed::Os::Linux => "code-stats-ls",
-            zed::Os::Windows => "code-stats-ls.exe",
-        };
-
-        if let Some(path) = worktree.which(&binary_name) {
+        if let Some(path) = worktree.which("code-stats-ls") {
             return Ok(path);
         }
 
@@ -40,6 +34,7 @@ impl CodeStatsExtension {
             },
         )?;
 
+        let (platform, arch) = zed::current_platform();
         let target_triple = format!(
             "code-stats-ls-{arch}-{os}",
             arch = match arch {
@@ -70,11 +65,17 @@ impl CodeStatsExtension {
 
         let version_dir = format!("code-stats-ls-{}", release.version);
 
-        // Windows have a different archive structure, as documented at:
-        // https://axodotdev.github.io/cargo-dist/book/artifacts/archives.html#archive-contents
-        let binary_path = match platform {
-            zed::Os::Windows => format!("{version_dir}/{}", binary_name),
-            _ => format!("{version_dir}/{target_triple}/{}", binary_name),
+        let (binary_path, file_type) = match platform {
+            zed::Os::Mac | zed::Os::Linux => (
+                format!("{version_dir}/{target_triple}/code-stats-ls"),
+                zed::DownloadedFileType::GzipTar,
+            ),
+            // Windows uses a different archive structure, as documented in:
+            // https://axodotdev.github.io/cargo-dist/book/artifacts/archives.html#archive-contents
+            zed::Os::Windows => (
+                format!("{version_dir}/code-stats-ls.exe"),
+                zed::DownloadedFileType::Zip,
+            ),
         };
 
         if !fs::metadata(&binary_path).map_or(false, |stat| stat.is_file()) {
@@ -83,17 +84,8 @@ impl CodeStatsExtension {
                 &zed::LanguageServerInstallationStatus::Downloading,
             );
 
-            let download_type = match platform {
-                zed::Os::Windows => zed::DownloadedFileType::Zip,
-                _ => zed::DownloadedFileType::GzipTar,
-            };
-
-            zed::download_file(
-                &asset.download_url,
-                &version_dir,
-                download_type,
-            )
-            .map_err(|err| format!("failed to download file: {err}"))?;
+            zed::download_file(&asset.download_url, &version_dir, file_type)
+                .map_err(|err| format!("failed to download file: {err}"))?;
 
             let entries = fs::read_dir(".")
                 .map_err(|err| format!("failed to list working directory {err}"))?;

